src/name/name.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2015-2020 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR
+// ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+// ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+// OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+use crate::DnsNameRef;
+
+use super::ip_address::{self, IpAddressRef};
+
+/// A DNS name or IP address, which borrows its text representation.
+#[derive(Debug, Clone, Copy)]
+pub enum DnsNameOrIpRef<'a> {
```

Comment:
Hmm, I don't think this is a great name? It is factually correct, of course, but I think we should express the domain how the values get used. Does the web PKI have a good name for this? I think on the rustls side we're calling it `ServerName`, but maybe we want to use it for clients here, too?

Reply:
I am find renaming this to something more meaningful.

Reply:
Been thinking about this, seems like `SubjectName` might be a decent option?

Reply:
What about `SubjectNameRef`, to keep a similar naming style?

Reply:
Addressed in https://github.com/rustls/webpki/pull/21 (https://github.com/rustls/webpki/pull/21/commits/1ed363434042fcfac9a8b27b2a432bc987fac71a)

Reply:
Sure, we want to keep the Ref suffix.